### PR TITLE
TextureOffset not support sampler2DArrayShadow sampler until 430.

### PR DESCRIPTION
**Pull request by ZhiqianXia**
Hi ,
glslang should report an error about textureOffset , because textureOffset not support sampler2DArrayShadow sampler until 430.

shader

#version 300 es
precision mediump float;
in highp vec4 dEQP_Position;
uniform mediump sampler2DArrayShadow s;
void main()
{
                gl_Position = vec4(textureOffset(s, vec4(0), ivec2(0)));
                gl_Position = dEQP_Position;
}
Reference:

https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.4.20.pdf
https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.4.30.pdf

### CODE REVIEW

File: glslang/MachineIndependent/ParseHelper.cpp [hunk 1]
```diff
@@ -2191,6 +2191,13 @@ void TParseContext::builtInOpCheck(const TSourceLoc& loc, const TFunction& fnCan
                               "[gl_MinProgramTexelOffset, gl_MaxProgramTexelOffset]");
                 }
             }
+
+            if (callNode.getOp() == EOpTextureOffset && version <= 420) {
```
Comment (greg-lunarg) [inline glslang/MachineIndependent/ParseHelper.cpp:2195]: Sorry, but I just noticed this.

The logic would be cleaner and the error messaging more appropriate if the version test was removed here and done below.

File: glslang/MachineIndependent/ParseHelper.cpp [hunk 2]
```diff
@@ -2191,6 +2191,13 @@ void TParseContext::builtInOpCheck(const TSourceLoc& loc, const TFunction& fnCan
...
+                TSampler s = arg0->getType().getSampler();
+                if (s.is2D() && s.isArrayed() && s.isShadow())
+                    error(loc, "TextureOffset not support the sampler2DArrayShadow : ", "sampler",
```
Comment (greg-lunarg) [inline glslang/MachineIndependent/ParseHelper.cpp:2198]: Replace the error call here with the following logic:

if (isEsProfile())
    error(loc, "TextureOffset does not support sampler2DArrayShadow : ", "sampler", "ES Profile");
else if (version <= 420)
    error(loc, "TextureOffset does not support sampler2DArrayShadow : ", "sampler", "version <= 420");

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
Sorry, some additional improvements requested.

You can make the changes on your current local branch, squash to one commit with rebase -i HEAD~2, and then force push the branch with git push -f <your fork> <your branch>.

File: glslang/MachineIndependent/ParseHelper.cpp (hunk 1 above)
Comment (ZhiqianXia) [reply glslang/MachineIndependent/ParseHelper.cpp]: Thank your commits.

**Comment by greg-lunarg** [comment]
I just pushed a commit onto your Feature2 branch. Your new test needed to be added to the master test list. And the compare file needed to be updated accordingly.

With this change, if it passes CI, all my concerns will be addressed and I can merge.

**Review by greg-lunarg — APPROVED** [review]
With my latest commit, this PR should be good to merge.

**Comment by ZhiqianXia** [comment]
@greg-lunarg Thank you for your commit!

**Pull request merged**